pylibui/libui/entry.py

```diff
@@ -0,0 +1,138 @@
+"""
+ Python wrapper for libui.
+
+"""
+
+import ctypes
+from . import clibui
+
+
+class uiEntry(ctypes.Structure):
+    """Wrapper for the uiEntry C struct."""
+
+    pass
+
+
+def uiEntryPointer(obj):
+    """
+    Casts an object to uiEntry pointer type.
+
+    :param obj: a generic object
+    :return: uiEntry
+    """
+
+    return ctypes.cast(obj, ctypes.POINTER(uiEntry))
+
+
+# - char *uiEntryText(uiEntry *e);
+def uiEntryText(entry):
+    """
+    Returns the text of the entry.
+
+    :param entry: uiEntry
+    :return: string
+    """
+
+    clibui.uiEntryText.restype = ctypes.c_char_p
+    text = clibui.uiEntryText(entry)
+
+    return text.decode()
+
+
+# - void uiEntrySetText(uiEntry *e, const char *text);
+def uiEntrySetText(entry, text):
+    """
+    Sets the text of the entry.
+
+    :param entry: uiEntry
+    :param text: string
+    :return: None
+    """
+
+    clibui.uiEntrySetText(entry, bytes(text, 'utf-8'))
+
+
+# - void uiEntryOnChanged(uiEntry *e, void (*f)(uiEntry *e, void *data), void *data);
+def uiEntryOnChanged(entry, callback, data):
+    """
+    Executes the callback function on entry change.
+
+    :param entry: uiEntry
+    :param callback: function
+    :param data: data
+    :return: reference to C callback function
+    """
+    c_type = ctypes.CFUNCTYPE(
+        ctypes.c_int, ctypes.POINTER(uiEntry), ctypes.c_void_p)
+    c_callback = c_type(callback)
+
+    clibui.uiEntryOnChanged(entry, c_callback, data)
+
+    return c_callback
+
+
+# - int uiEntryReadOnly(uiEntry *e);
+def uiEntryReadOnly(entry):
+    """
+    Returns whether the entry is read only or not.
+
+    :param entry: uiEntry
+    :return: string
+    """
+
+    return bool(clibui.uiEntryReadOnly(entry))
```

Comment:
Yes you are right. The reason I did not do it is simply because I wrote the code before the issue was submitted. (24 Jun)
I can fix it on this PR; however I already did another branch -which I will submit in a PR after this one- [which removes all the casts in `pylibui/libui/*.py` files](https://github.com/superzazu/pylibui/commit/bb1dcfe7510c4f57ede1ca10ecc4c8f7aa2b9953). It's which one you prefer :-)

Concerning the "int" instead of "string", you are right